platform-common-date/src/main/java/com/softicar/platform/common/date/DayParser.java

```diff
@@ -32,15 +32,38 @@ public DayParser(String text) {
 		this.text = text;
 	}
 
-	public Optional<Day> parse() {
+	/**
+	 * Parses the given text into a {@link Day}.
+	 *
+	 * @return the parsed {@link Day} object (never <i>null</i>)
+	 * @throws IllegalDateSpecificationException
+	 *             if the given text could not be parsed
+	 */
+	public Day parseOrThrowIfInvalid() {
```

Comment:
We could consider removal of the `IfInvalid` suffix. Why else would it throw?

Reply:
hmm, ok, I copied this name suffix from the inputs but you are right, it is not needed here